### PR TITLE
W-16560050-AsyncAPI-Avro-support-CP

### PR DESCRIPTION
**Pull request by Crispy-Salesforce**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+APIkit for AsyncAPI supports Avro, a data serialization system that helps you define the format for specifying the structure and encoding of messages. This module shows you the structure of messages in the design phase. APIkit for AsyncAPI also supports xref:studio::datasense-concept.adoc[DataSense] of messages with Avro schema.
+
+== Schema Definitions
+
+The Avro spec defines its format by giving the value of `schemaFormat` as `application/vnd.apache.avro`.
+
+The Avro spec can describe the schema inline:
+
+----
+ schemas:
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: Please use this instead
```
    publish:
      summary: inform details about person
      operationId: lightMeasured
      message:
        schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
        payload:
          name: lightMeasured
          type: record
          fields:
            - name: lumens
              type: int
              example: 134
            - name: sentAt
              type: string
              logicalType: timestamp-millis
              default: 10
              example: 1985-04-12T23:20:50.52Z

```

Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: OK! Thanks!

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      name: lightMeasured
+      type: record
+      fields:
+        - name: lumens
+          type: int
+          example: 134
+        - name: sentAt
+          type: string
+          logicalType: timestamp-millis
+          default: 10
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: sorry, remove this `default`

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+The Avro spec can also refer to an external document that describes the schema in the Avro format:
+
+----
+channels:
+ smartylighting.streetlights.1.0.event.{streetlightId}.monitor:
+   publish:
+     message:
+       schemaFormat: application/vnd.apache.avro;version=1.9.0
+       payload:
+         $ref: lumens.avsc'
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: sorry, another change. Just learned how schema fragments published on exchange are referred. So modify this line as 
`$ref: /exchange_modules/83589b9b-6630-4c28-985d-174ee9172661/lumenschema/1.0.0/lumens.json'`

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+----
+
+The `lumens.avsc` file provides the specification in the Avro schema.
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: Please change as `lumens.json`
Exchange expects file extension to be json

**Review by aby-philip-sf — COMMENTED** [review]
Sorry, just learned about some nuances regarding how schema is published in exchange. 
Few more updates please.

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc (hunk 2 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: Please, remember that you can add a suggestion and edit the doc yourself. 
Once done, I can commit to your suggestion.
Thanks!

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc (hunk 3 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: Please, remember that you can add a suggestion and edit the doc yourself.
Once done, I can commit to your suggestion.
Thanks!

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc (hunk 4 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]: Please, remember that you can add a suggestion and edit the doc yourself.
Once done, I can commit to your suggestion.
Thanks!

**Review by aby-philip-sf — APPROVED** [review]

File: modules/ROOT/pages/apikit-asyncapi-avro-support.adoc [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+----
+channels:
+ smartylighting.streetlights.1.0.event.{streetlightId}.monitor:
+   publish:
+     message:
+       schemaFormat: application/vnd.apache.avro;version=1.9.0
+       payload:
+         $ref: /exchange_modules/83589b9b-6630-4c28-985d-174ee9172661/lumenschema/1.0.0/lumens.json'
+----
```
Comment (Cristian-Venticinque) [inline modules/ROOT/pages/apikit-asyncapi-avro-support.adoc]:
```suggestion
The Avro spec supports describing the schema inline or referencing an external document that describes the schema in the Avro format:

* Inline definition
+
----
publish:
  summary: inform details about person
  operationId: lightMeasured
  message:
    schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
    payload:
      name: lightMeasured
      type: record
      fields:
        - name: lumens
          type: int
          example: 134
        - name: sentAt
          type: string
          logicalType: timestamp-millis
          10
          example: 1985-04-12T23:20:50.52Z
----

* Reference to an external definition
+
----
channels:
 smartylighting.streetlights.1.0.event.{streetlightId}.monitor:
   publish:
     message:
       schemaFormat: application/vnd.apache.avro;version=1.9.0
       payload:
         $ref: /exchange_modules/83589b9b-6630-4c28-985d-174ee9172661/lumenschema/1.0.0/lumens.json'
----
```

Optional: Suggestion to reduce wording and try to show better that these are two ways of defining the spec. If you apply this suggestion, please confirm in a local build that the bulleted list renders as intended.

**Review by Cristian-Venticinque — APPROVED** [review]
Approved with one optional suggestion.

**Pull request merged**